Segmented Control: playing with colors

rickybanister opened the pull request:
I updated the segmented control to use blue-medium (our more standard 'this thing is selected' color) and blue-wordpress/link-highlight for the non-selected options since they are clickable link-like things. It also much more closely matches the iOS segmented control that inspired it in the first place.

**Before:**
<img width="696" alt="screen shot 2016-02-02 at 3 47 26 pm" src="https://cloud.githubusercontent.com/assets/437258/12763778/4ecb4f94-c9c4-11e5-99bf-6a7dbcdf040b.png">

**After:**
<img width="700" alt="screen shot 2016-02-02 at 3 34 32 pm" src="https://cloud.githubusercontent.com/assets/437258/12763783/53b66e26-c9c4-11e5-8070-9a62a820bc6e.png">

This might benefit from some additional work on tabbing through options and focus/hover styles, but I think it's much more friendly now and matches our iOS app more closely.

cc @mtias @drw158 @andrewspittle (who originally suggested updating the colors since they didn't quite match our form field colors nor much other interface)

drw158 commented:
Stylistically I like this a little bit better

![screen shot 2016-02-02 at 3 13 53 pm](https://cloud.githubusercontent.com/assets/618551/12764496/bd1a5094-c9bf-11e5-8d66-837e39ba54e5.png)

The border could be a little bit lighter.

Something about the different shades of blue in your screenshot throw me off a little bit.

rickybanister commented:
We could go your route but use the form-field border colors:

`lighten( $gray, 20% );`
and for hover `lighten( $gray, 10% );`

At least that would match other borders around town.

I think the main improvement is having the selected item be the blue-medium.

rickybanister commented:
I tried out your suggestion @drw158 

What do people think:

<img width="692" alt="screen shot 2016-02-04 at 5 43 16 pm" src="https://cloud.githubusercontent.com/assets/437258/12832276/0245e2d4-cb67-11e5-9248-dcae5a85f554.png">

('Discouraged' is hovered)

drw158 commented:
This looks good to me. It's readable, and it's clear which one is selected.

I was imagining that the hover would be identical to the selected state. Hovering the selected state wouldn't change the color, since nothing happens when you click it.

mtias commented:
I like this. Segmented control is used in quite a few places, notably the editor mode tabs. Does it work as expected in all cases?

mtias commented:
Editor looks bad:

![image](https://cloud.githubusercontent.com/assets/548849/12842506/881ed06e-cbf3-11e5-8c88-c78c87c59ef3.png)

mtias commented:
Maybe worth retaining the old color behaviour with a `secondary` prop matching what we do with buttons?

rickybanister commented:
Which would be the secondary style? I think that makes sense, I might want a little help making that happen though.

mtias commented:
Sure, ping me and we'll work it through.

rickybanister commented:
I fixed the editor regression thanks to @mtias adding a `primary` prop. I also changed the default style slightly to better match our form fields and other elements of calypso. We would need to copy that over to notifications if we decide we like it.

Here's where I ended up:

<img width="498" alt="screen shot 2016-02-12 at 7 12 11 pm" src="https://cloud.githubusercontent.com/assets/437258/13024008/082c3b42-d1bd-11e5-80d9-1b6f2e4935f6.png">

My issue with the old style was twofold: using a gray background made the selected item feel like it was receding at times, and the `$gray` border was darker than we use elsewhere. I feel we're now being more consistent.

drw158 commented:
I like it. We should use the new primary style as much as possible, and maybe eventually phasing out the secondary style imo. Thanks for doing this, Rick.

I can copy over the new style to Notifications.

mtias commented:
Looks good to me.
